medusa/__main__.py

```diff
@@ -1247,6 +1258,27 @@ def path_leaf(path):
                 folder_path = os.path.join(cache_folder, name)
                 helpers.remove_folder(folder_path)
 
+    def migrate_python_version(self, new_version):
+        """
+        Perform some cleanup in case we switch from python version.
+
+        It's possible to switch from python version 2 to 3 or visa versa.
+        In that case we might wanna run some sanitary actions, to make sure everything keeps working.
+        :param new_version: currently started python version
+        :type new_version: int
+        """
+        # TODO: Leaving this here as a template for when we upgrade merge the python3 changes.
+        if new_version == 3:
+            pass
+        else:
+            pass
+
+        app.PYTHON_VERSION = new_version
+
+        # save config
+        logger.info(u'Saving config file to disk')
+        self.save_config()
```

Comment:
I don"t think this is needed? It's not done anywhere else and everything seems to be saved on shutdown.

Reply:
Well. No in not all cases its saved on shutdown nor restart

Reply:
How so? It looks like the config is always saved on shutdown, see: https://github.com/pymedusa/Medusa/blob/3644aecf38a6689f1cfb76a8d0833b5d9376988d/medusa/__main__.py#L2158-L2168

Reply:
okay, will depend on that